lib/src/asset_graph/node.dart

```diff
@@ -16,32 +15,56 @@ class AssetNode {
 
   AssetNode(this.id);
 
+  factory AssetNode.deserialize(List serializedNode) {
```

Comment:
I'll put in a quick plug for protos here. They're not perfect but you get backwards-compatible serialization, composition, etc. for free and their implementations will improve without you doing anything. I think it's worth the up front cost.